src/common/libidset/idset.c

```diff
@@ -124,6 +157,124 @@ int idset_set (struct idset *idset, unsigned int id)
     return 0;
 }
 
+int idset_range_set (struct idset *idset, unsigned int lo, unsigned int hi)
+{
+    unsigned int id;
+
+    if (!idset || !valid_id (lo) || !valid_id (hi)) {
+        errno = EINVAL;
+        return -1;
+    }
+    if (hi < lo) {
+        id = hi;
+        hi = lo;
+        lo = id;
+    }
+    if (idset_grow (idset, hi + 1) < 0)
+        return -1;
+    for (id = lo; id <= hi; id++)
+        vebput (idset->T, id);
+    return 0;
+}
+
+int idset_clear (struct idset *idset, unsigned int id)
+{
+    if (!idset || !valid_id (id)) {
+        errno = EINVAL;
+        return -1;
+    }
+    vebdel (idset->T, id);
+    return 0;
+}
+
+int idset_range_clear (struct idset *idset, unsigned int lo, unsigned int hi)
+{
+    unsigned int id;
+
+    if (!idset || !valid_id (lo) || !valid_id (hi)) {
+        errno = EINVAL;
+        return -1;
+    }
+    if (hi < lo) {
+        id = hi;
+        hi = lo;
+        lo = id;
+    }
+    for (id = lo; id <= hi && id < idset->T.M; id++)
+        vebdel (idset->T, id);
+    return 0;
+}
+
+bool idset_test (const struct idset *idset, unsigned int id)
+{
+    if (!idset || !valid_id (id))
+        return false;
+    return (vebsucc (idset->T, id) == id);
+}
+
+unsigned int idset_first (const struct idset *idset)
+{
+    unsigned int next = IDSET_INVALID_ID;
+
+    if (idset) {
+        next = vebsucc (idset->T, 0);
+        if (next == idset->T.M)
+            next = IDSET_INVALID_ID;
+    }
+    return next;
+}
+
+
+unsigned int idset_next (const struct idset *idset, unsigned int prev)
+{
+    unsigned int next = IDSET_INVALID_ID;
+
+    if (idset) {
+        next = vebsucc (idset->T, prev + 1);
+        if (next == idset->T.M)
+            next = IDSET_INVALID_ID;
+    }
+    return next;
+}
+
+size_t idset_count (const struct idset *idset)
+{
+    unsigned int id;
+    size_t count = 0;
+
+    if (!idset)
+        return 0;
+    id = vebsucc (idset->T, 0);
+    while (id < idset->T.M) {
+        count++;
+        id = vebsucc (idset->T, id + 1);
+    }
+    return count;
+}
+
+bool idset_compare (const struct idset *idset1,
+                    const struct idset *idset2)
+{
+    unsigned int id;
+
+    if (!idset1 || !idset2)
+        return false;
+
```

Comment:
minor thought, would it be a net win to do some obvious checks before iterating through everything below?  I know ```vebsize()``` is costly and perhaps not worth it.  T.M != T.M?  I don't know the library that well, so maybe it's not a net win.

Reply:
`T.M` is the number of available slots in the set, not filled slots.  `vebsize()` is the size in bytes of the data structure, not related to filled slots.  So I don't think either of those is helpful here.  

In looking at this, I realized there are no users of `idset_compare()` so hey, let's just get rid of it.